docker/cloudbuild.yaml

```diff
@@ -32,13 +40,20 @@ steps:
   - '--file=Dockerfile-17.06.1'
   - '.'
   wait_for: ['17.06.1']
+- name: 'gcr.io/$PROJECT_ID/docker:17.12.0'
+  args:
+  - 'build'
+  - '--file=Dockerfile-17.12.0'
+  - '.'
+  wait_for: ['17.12.0']
+
 # Tests for future supported versions of docker builder go here.
 
 # Tag the latest version as :latest. We use gcr.io/cloud-builders/docker here
 # and not gcr.io/$PROJECT_ID/docker because the latter may not yet exist.
 - name: 'gcr.io/cloud-builders/docker'
-  args: ['tag', 'gcr.io/$PROJECT_ID/docker:17.06.1', 'gcr.io/$PROJECT_ID/docker']
-  wait_for: ['17.06.1']
+  args: ['tag', 'gcr.io/$PROJECT_ID/docker:17.12.0', 'gcr.io/$PROJECT_ID/docker']
```

Comment:
We can't tag `17.12` as `:latest` until the server upgrade is complete, and until we've tested that the new client version doesn't break anything.

`17.12` will be broken until the server upgrade, but simply building+pushing `17.12` is fine, since nobody is forced to use it (until it's tagged `:latest`).